### DISCUSSION TITLE
use extension also with e2e files

### DISCUSSION THREAD

**Discussion by Sevenhero** [orta/vscode-jest#812] [Ideas]
This is the first time I ever wrote something here in github :)
Hello :)  I really like your extension.
I run e2e tests with jest via terminal so in the  "package.json"
"test:e2e": "jest --config ./test/jest-e2e.json"
my question would be : can I somehow configure that the extension also picks up these files?
their naming is like *.e2e-spec.ts 

thank you :)

**Comment by connectdotz** [comment]
@Sevenhero thanks for trying the extension. If you just want the extension  to run `test:e2e`, you can add [jest.jestCommandLine](https://github.com/jest-community/vscode-jest#jestcommandline) in your settings.json, such as: `"jest.jestCommandLine": "yarn test:e2e"`. Please let me know if I misunderstood your request.

**Reply by Sevenhero** [reply]
hello @connectdotz thank you for your fast answer.
what I mean is: in spec files the extension shows the symbols on the left of the file for debugging.
i would like to start there also the e2e-spec files ... but than it would have to use another config file so I'm not sure if this is even possible

**Reply by pedro040529** [reply]
One option could be to rename your e2e test files to match the naming convention for spec files (*.spec.ts). This would allow the extension to recognize and display the symbols for these files as well. Another option could be to modify the configuration for the VS Code extension for Jest to recognize your e2e test files with a different naming convention.

I hope this helps!

**Comment by connectdotz** [comment]
hi @Sevenhero this is an interesting idea, unfortunately right now we only support 1 jest run per workspace. You could maybe hack around to create a separate workspace with a different config... 🤔

**Comment by connectdotz** [comment]
to clarify, I mean to create a [multiroot workspace](https://code.visualstudio.com/docs/editor/multi-root-workspaces) with a separate folder (let's call it "e2e"), then you can customize this 'e2e' folder with something like:
```json
// in e2e/.vscode/settings.json
{
  ...
  "jest.rootPath":"../",
  "jest.jestCommandLine": "yarn test:e2e"
}
```

You might need to adjust something else, just some idea if you want to try... let me know if you got stuck.

**Reply by dale-vendia** [reply]
Hey @connectdotz -- can you confirm (as I'm unable to tell).

Does the above approach:
1/ create a new workspace folder that points to the e2e tests? 
2/ the vscode-jest plugin is somehow able to pick up the e2e settings.json because it's the closest to the test file?

It's really frustrating that jest doesn't have a better answer to conditionally defining set-up files per folder, and it becomes challenging to do integration / e2e testing + unit testing.

**Comment by connectdotz** [comment]
@dale-vendia 

You should create the workspace folder for the directory where the e2e `settings.json` is. It has nothing to do with the test files.  In that `settings.json` you can then customize the e2e `rootPath` or `jestCommandLine` etc to help the extension find test files/config/command in that workspace folder.

Hope this help.
